### ISSUE TITLE
Get rid of a,b,c,d one-letter codes for variables

### ISSUE THREAD

**Issue by targos** [cheminfo/cheminfo-types#86]

**Comment by santimirandarp** [comment]
And the defaults `x`, `y`, as well ? What would you add instead, `[varName:string]` ? 

The "real key name" is being now found from the sub-key `label` now which seems inefficient. But the only problem is that variables aren't easy to camel-case (some end up being non-human readable.) 

Also, the name may be `E/s` and `E/h`, where the second letter is a unit, so to make it work the full `E/x` name should be included as a key.

Would be nice to know what is your idea.

**Comment by targos** [comment]
My idea was to use an array instead of an object, so we don't need to come up with arbitrary keys.

**Comment by santimirandarp** [comment]
Nice idea. So [this key here](https://github.com/cheminfo/cheminfo-types/blob/946936b3dd5d4af1e1a4298c9cc9bf317b470a45/src/measurements/common/MeasurementXY.ts#L15) would become an array of MeasurementVariables, with `monotonic` as well. Correct?

idk what would be preferred for flagging default `x` and `y`, or whether that should be here or may be better elsewhere.

**Comment by targos** [comment]
> So [this key here](https://github.com/cheminfo/cheminfo-types/blob/946936b3dd5d4af1e1a4298c9cc9bf317b470a45/src/measurements/common/MeasurementXY.ts#L15) would become an array of MeasurementVariables, using monotonic as well. Correct?

Correct, although I don't understand why monotonic has to do with this.

> idk what would be preferred for flagging default x and y

We could just say that the array may be sorted in a way that the first two elements are the preferred x and y.

> This will change some of the parsers.

Maybe, but I think this would be an indication that the parsers are doing something wrong. IMO parsers should return a JavaScript object representing the file that they are parsing, not something transformed to comply to the measurement types defined here.

**Comment by santimirandarp** [comment]
> why monotonic has to do with this.

I had just read the other issue, I was just thinking what modifications will be included in the new type.

Thanks for the explanations. I agree, just couldn't really find out how to do it better. 

But it does seem - to me at least - appropriate to make the parsers somewhat uniform using similar types.
